### DISCUSSION TITLE
Black borders below rounded corners in GTK apps

### DISCUSSION THREAD

**Discussion by abrarbinkabir** [qtile/qtile#4477] [Q&A]
I am not sure if this question is totally related to Qtile; I thought some of you guys might have faced this in Qtile and might have solution. I have installed **picom** from **ArchLinux extra repo** and set corner radius to 7. My GTK theme (Catppuccin Latte) has a greater corner radius than what I've set in picom. I think that's the reason I get these annoying black borders below the corners. Do you have any idea how to solve it? Thanks in advance.

![HPjnPO5m_o](https://github.com/qtile/qtile/assets/114284302/82892d6e-397c-4578-9fff-a07868432933)

**Comment by jrwrigh** [comment]
I'm honestly not sure. But if the dropdown menu is a separate X11 window (which it'd have to be for Picom to have control over it if I understand it correctly), you can try and get the window attributes and then set specific settings for that kind of window in Picom. 

I'd normally get those window attributes via `xprop`, but I'm not sure how feasible that is for a drop-down menu. But it's also possible that there isn't enough unique X11 identifiers to allow limiting a subset of Picom rules to them.

**Reply by abrarbinkabir** [reply]
The problem is that when I switch to terminal to run **xprop**, those dropdowns collapse. I am not sure how would I get their class.

**Reply by jrwrigh** [reply]
Yep, that's what I was afraid of. Unfortunately I don't know a way around that. Maybe some kind of X11 logging might do the trick? No idea how to do that though.

**Reply by jrwrigh** [reply]
Actually, maybe I have something? The picom sample config has [this:](https://github.com/yshui/picom/blob/751f30578e352f6cd37fc63cbf4643e47ab3587a/picom.sample.conf#L140-L153)
```conf
#################################
#           Corners             #
#################################


# Sets the radius of rounded window corners. When > 0, the compositor will
# round the corners of windows. Does not interact well with
# `transparent-clipping`.
corner-radius = 0


# Exclude conditions for rounded corners.
rounded-corners-exclude = [
  "window_type = 'dock'",
  "window_type = 'desktop'"
];
```

and [this:](https://github.com/yshui/picom/blob/751f30578e352f6cd37fc63cbf4643e47ab3587a/picom.sample.conf#L392-L395)
```conf
# 'WINDOW_TYPE' is one of the 15 window types defined in EWMH standard:
#     "unknown", "desktop", "dock", "toolbar", "menu", "utility",
#     "splash", "dialog", "normal", "dropdown_menu", "popup_menu",
#     "tooltip", "notification", "combo", and "dnd".
```

So for the `exclude`, maybe you could add in something like 
```
rounded-corners-exclude = [
  "window_type = 'dock'",
  "window_type = 'desktop'"
  "window_type = 'menu'"
  "window_type = 'dropdown_menu'"
  "window_type = 'popup_menu'"
];
```?

[This Picom issue](https://github.com/yshui/picom/issues/552#issuecomment-736602342) uses all three of the "`menu`" window types in a different exclude statement. So maybe that'll work here too?

There's also  a `detect_rounded_corners` [option](https://github.com/yshui/picom/blob/751f30578e352f6cd37fc63cbf4643e47ab3587a/picom.sample.conf#L238-L242), which might be worth toggling? I currently have mine set to `false`, but I also don't have any rounded corners.
